test/integration/api/integrations_controller_test.rb

```diff
@@ -170,4 +170,15 @@ def test_edit
     get edit_admin_service_integration_path(service_id: service.id)
     assert_response :not_found
   end
+
+  def
+    service = FactoryBot.create(:simple_service, account: @provider)
+    get admin_service_integration_path(service_id: service)
+    service.stubs(:oauth?).retuns(true)
```

Comment:
This will not work, for 2 reasons:
1. You are stubbing AFTER the request, so the request is already done without being stubbed.
2. Being an integration test, it picks this service but a different instance, so it won't use this stub.